### PR TITLE
Fix dead links in docs

### PR DESCRIPTION
**Pull request by kevinjqliu**
#324 added Github action to check link html links in docs. There are a few dead links in the most recent run. 
https://github.com/kevinjqliu/iceberg-python/actions/runs/8124843259/job/22207404913

This PR corrects the dead links or disables checking for those particular cases. 

```
ERROR: 1 dead links found!
[✖] https://iceberg.apache.org/catalog/ → Status: 404

ERROR: 6 dead links found!
[✖] https://10.0.19.25/ → Status: 0
[✖] http://my.proxy.com:8080 → Status: 0
[✖] http://localhost/ → Status: 0
[✖] http://0.0.0.0:4443 → Status: 0
[✖] https://rest-catalog/ws → Status: 0
[✖] https://auth-service/cc → Status: 0

ERROR: 1 dead links found!
[✖] reference/ → Status: 400
```

Following instructions in 
https://github.com/gaurav-nelson/github-action-markdown-link-check

### CODE REVIEW

File: mkdocs/docs/SUMMARY.md [hunk 1]
```diff
@@ -26,6 +26,6 @@
 - Releases
     - [Verify a release](verify-release.md)
     - [How to release](how-to-release.md)
-- [Code Reference](reference/)
+- [Code Reference](reference/) <!-- markdown-link-check-disable-line -->
```
Comment (HonahX) [inline mkdocs/docs/SUMMARY.md:29]: I tried to build the site locally but received the following error
```
mkdocs_literate_nav.parser.LiterateNavParseError: Expected no text after <a href="reference/">Code Reference</a>, but got ' <!-- markdown-link-check-disable-line -->'.
```
Seems we also need to use `<!-- markdown-link-check-disable -->`/`<!-- markdown-link-check-enable-->` here?

Comment (kevinjqliu) [reply mkdocs/docs/SUMMARY.md]: good catch! I was able to verify that the updated PR can build the site successfully

```
cd mkdocs
mkdocs build --strict                                               
```

**Review by HonahX — APPROVED** [review]
LGTM! Thanks @kevinjqliu 

BTW, I noticed that the markdown-link-check isn't automatically run with this PR. Is it intended or we need to somehow update the workflow yml?

**Comment by kevinjqliu** [comment]
@HonahX that is weird. The github action is currently configured to run when there are changes to `mkdocs/**`, which should definitely include this PR.
https://github.com/apache/iceberg-python/blob/main/.github/workflows/check-md-link.yml#L6

I see it ran for the latest commit in my github fork
https://github.com/kevinjqliu/iceberg-python/actions/runs/8133371492/job/22224875853

**Comment by kevinjqliu** [comment]
In fact, I'm not sure if the `Check Markdown links` github action has ever ran for this repo
https://github.com/apache/iceberg-python/actions/workflows/check-md-link.yml

**Comment by kevinjqliu** [comment]
Looks like its related to this 
https://stackoverflow.com/questions/61989951/github-action-workflow-not-running

CI and integration test github actions both specifically reference the `main` branch
https://github.com/apache/iceberg-python/blob/main/.github/workflows/python-ci.yml#L25

I can open an issue for this

**Comment by kevinjqliu** [comment]
I just pushed the change following the same format as https://github.com/apache/iceberg-python/blob/main/.github/workflows/python-ci.yml#L25

this PR should now run the `check md link` github action 

@HonahX

**Comment by HonahX** [comment]
Thanks for fixing the CI config!

**Pull request merged**